### PR TITLE
DBZ-4896 Allow `ON` as an expression value

### PR DESCRIPTION
**Pull request by vjuranek**
`ON` can be used as a value for e.g. some server system variables
like unique_checks [1].

[1] https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_unique_checks

### CODE REVIEW

**Comment by vjuranek** [comment]
Antlr grammar PR: https://github.com/antlr/grammars-v4/pull/2554

**Comment by jpechane** [comment]
DOes it support `set unique_checks=off`?

**Comment by vjuranek** [comment]
> DOes it support `set unique_checks=off`?

Yes, as `OFF` is not in lexer, so it's `off` works without any additional fixes.

File: debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4 [hunk 1]
```diff
@@ -1717,8 +1717,8 @@ uninstallPlugin
 //    Set and show statements
 
 setStatement
-    : SET variableClause ('=' | ':=') expression
-      (',' variableClause ('=' | ':=') expression)*                             #setVariable
+    : SET variableClause ('=' | ':=') (expression | ON)
```
Comment (gunnarmorling) [inline debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4:1720]: It works, but wouldn't it make more sense to make `ON` part of `expression`? Or is it really _only_ allowed exactly here?

**Review by gunnarmorling — COMMENTED** [review]
One question inline, @vjuranek.

File: debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4 (hunk 1 above)
Comment (vjuranek) [reply debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4:1720]: I'm not MySQL expert to be able to say it for sure that this is the only place where it's allowed. And at the same time I'm not able to say for sure that adding it into `expression`, which is widely used, won't cause any issues as `ON` can have different meaning in some other circumstances/expressions. So this seems to me to be more safe change.

**Comment by vjuranek** [comment]
Also updated example with `off` case - for clarity and also to fail fast in the future if `off` is added into the lexer.

File: debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4 (hunk 1 above)
Comment (gunnarmorling) [reply debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4:1720]: Ok, makes sense, let's do it the proposed way then.

For some context, in general, we don't care about the semantics of these grammar parts at all from a Debezium perspective (just need to be able to parse it _somehow_, so we can properly parse what we're actually after, which is table DDL). So we can sometimes take shortcuts, though the aspiration still is to have a grammar that makes sense (not the least given we feed back those changes upstream to Antlr).

Comment (vjuranek) [reply debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4:1720]: Thanks for explanation - this is what I thought, that for Debezium it doesn't matter, at as I submitted this change also to Antlr, I don't want to break anything. Someone more experience can improve it in upstream if needed.

**Pull request merged**